macrosynergy/management/utils/df_utils.py

```diff
@@ -300,6 +300,65 @@ def downsample_df_on_real_date(
     )
 
 
+def downsample_wide_df_on_real_date(
```

Comment:
Is this function only to be used in historic portfolio volatility? Historically we have focused on convert frequency? Do we not have a frequency conversion function already (we calculate monthly returns)?

Reply:
We do, but it works on the quantamental format. Ultimately (many many refactors later), we want to many of the backend method to use wide DFs, with an overload for QDFs

Reply:
Got it - we should consider then mapping it out and maybe have [1] QDF and [2] WDF as two distinct types.

Reply:
Yes. For now, `downsample_wide_df_on_real_date` and `downsample_df_on_real_date` exist for that.
We will most likely be adding more functionality around wide dataframes, or atleast context based switching between the two formats